src/Workspaces/Core/Portable/Storage/SQLite/v2/Interop/SqlStatement.cs

```diff
@@ -67,7 +69,36 @@ public Result Step(bool throwOnError = true)
         }
 
         internal void BindStringParameter(int parameterIndex, string value)
-            => _connection.ThrowIfNotOk(NativeMethods.sqlite3_bind_text(_rawStatement, parameterIndex, value));
+        {
+            // Attempt to stackalloc utf8 converted small strings to avoid lots of allocs.
+            //
+            // This is safe as sqlite will still copy our bytes over to its own internal memory (see
+            // conversation here: https://github.com/dotnet/roslyn/pull/51111#pullrequestreview-588169715)
+            // on the topic.  So it's fine that our own stackalloc'ed bytes will be gone when this function
+            // returns and our caller continues to interact with this SqlStatement instance.
+            var utf8ByteCount = Encoding.UTF8.GetByteCount(value);
```

Comment:
FWIW in [my version of this](https://dev.azure.com/devdiv/DevDiv/_git/VS.CloudCache/pullrequest/303830) I do a simple string-length check first to avoid adding the hit to encode the string here if the string itself has more characters than would be allowed in bytes.

Reply:
@AArnott good call!